pkg/polaris/polaris_test.go

```diff
@@ -748,3 +751,372 @@ func TestGcpPermissions(t *testing.T) {
+	if err!= nil {
+		t.Fatal(err)
+	}
+	fmt.Printf("%v\n", info)
+}
+
+// TestGetTaskchainInfo verifies that the SDK can fetch taskchain states
+func TestGetTaskchainInfo(t *testing.T) {
+	testClusterID := uuid.MustParse("a840e205-ac36-408d-9a12-7690769aaa88")
+	ctx := context.Background()
+	testServiceAccount := ServiceAccount{
+		ClientID:       "client|2YDsG5FIYWQ7OW8AiRqA2tPwQEwhHpfU",
+		ClientSecret:   "ZLeoEHA6dvDGciAO1bFIIWJmdjlLLawFa6WC8IdpBfgphecsUQYB4pAOKovQbt4e",
+		Name:           "kupatest",
+		AccessTokenURI: "https://demo.dev-017.my.rubrik-lab.com/api/client_token",
+	}
+
+	client, err := NewClient(ctx, &testServiceAccount, &polaris_log.DiscardLogger{})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	nss, err := client.K8s().ListK8sNamespace(ctx, testClusterID)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if len(nss) == 0 {
+		t.Fatal("No namespace found")
+	}
+
+	ns := nss[1] // get first namespace
+	info, err := client.K8s().TakeK8NamespaceSnapshot(ctx, ns.ID, "00000000-0000-0000-0000-000000000000")
+	if err!= nil {
+		t.Fatal(err)
+	}
+
+	for i := 0; i < 6; i++ {
+		tk, err := client.K8s().GetTaskchainInfo(ctx, info.TaskchainId, "snapshot-namespace")
+		if err == nil {
+			fmt.Printf("%v\n", tk)
+			return
+		}
+		if !strings.Contains(err.Error(), "Taskchain does not exist") ||
+			!strings.Contains(err.Error(), "State key not found in response") ||
+			!strings.Contains(err.Error(), "Taskchain still in wait state") {
+			fmt.Printf("%v : %v\n", info, err)
+			time.Sleep(15 * time.Second)
+		} else {
+			t.Fatal(err.Error())
+		}
+	}
+
+	t.Fatal("Task chain does not start")
+}
+
+// TestGetActivitySeriesConnection verifies that the SDK can fetch events
+func TestGetActivitySeriesConnection(t *testing.T) {
+	ctx := context.Background()
+	testServiceAccount := ServiceAccount{
+		ClientID:       "client|gYtGLLyhcp3rI02kkbIdZt7CMFzIhg54",
+		ClientSecret:   "JQfCRwuiiwzB_7Ibt1UIOcAT0wQTYnRNF2ikPD8aTaTOqbiYhfm8v7Lb6pY6NBis",
+		Name:           "manifest-unit-test",
+		AccessTokenURI: "https://manifest.dev-045.my.rubrik-lab.com/api/client_token",
+	}
+
+	client, err := NewClient(ctx, &testServiceAccount, &polaris_log.DiscardLogger{})
+	if err != nil {
+		t.Fatal(err)
+	}
+	ts := time.Now().Add(time.Duration(-2) * time.Hour).UTC()
+	as, err := client.K8s().GetActivitySeriesConnection(ctx, []string{"KuprNamespace"}, ts)
+	if err != nil {
+		t.Fatal(err)
+	}
+	fmt.Printf("%v\n", len(as))
+	fmt.Printf("%v\n", as)
+}
+
+// TestRestoreK8NamespaceSnapshot verifies that the SDK can restore a
+// namespace snapshot
+func TestRestoreK8NamespaceSnapshot(t *testing.T) {
+	testClusterID := uuid.MustParse("a840e205-ac36-408d-9a12-7690769aaa88")
+	ctx := context.Background()
+	testServiceAccount := ServiceAccount{
+		ClientID:       "client|2YDsG5FIYWQ7OW8AiRqA2tPwQEwhHpfU",
+		ClientSecret:   "ZLeoEHA6dvDGciAO1bFIIWJmdjlLLawFa6WC8IdpBfgphecsUQYB4pAOKovQbt4e",
```

Comment:
:exclamation:**Cycode**: Secret of type: 'Base64 Entropy' was found.
Severity: Medium <!-- cd5f5f0f-06d1-438f-8fee-ed458131aac7 -->